### ISSUE TITLE
A General and Adaptive Robust Loss Function

### ISSUE THREAD

**Issue by qtreh** [qtreh/papers#1]
Jonathan T. Barron
CVPR 2019
[arXiv:1701.03077 [cs.CV]](https://arxiv.org/abs/1701.03077)

**Goal**

The loss function is commonly used in machine learning to compute the distance between a real sample and the prediction achieved with the model. This error measure, called residual, is then used in the gradient computation to impact the weights update operation in a neural network during the backpropagation.

We want to customize the loss function to the dataset distribution. Some losses will penalize almost equally soft and hard outliers (such as the Geman-McClure, or Welsch losses) whereas others will compute a gradient that way more sensible as the residuals increase (such as L2 or Charbonnier losses). 

**Difficulties**

An increasing concern in deep learning is robustness, which is the ability of a model to adapt to outliers in the data. 

There are robust loss functions such as the L2 or Cauchy losses, but each is adapted to a certain nature of data, i.e how much you think outliers impact should be discarded. You could be aware that your data contains, sometimes, large outliers that are caused by a known glitch in a sensor for example. In that case you want a loss that has proven effective against hard outliers. But getting the right loss for this dataset will still require some test and try, where you have to benchmark your architecture against all the potential losses.

**Contributions**

![image](https://user-images.githubusercontent.com/8530703/81506920-eaadeb80-92f9-11ea-8690-09ea03b2945d.png)

The loss devised by the author encompasses a lot of other losses, only by changing the alpha parameter. The c parameter corresponds to the “bowl width”.

![image](https://user-images.githubusercontent.com/8530703/81506937-05806000-92fa-11ea-96a8-428286406b3b.png)

But rather than using this loss function, the author derived a probability distribution such that the negative log-likelihood (NLL) of its PDF is a shifted version of the loss function. The issue with the original loss is that the optimization would simply set alpha to be as small as possible to minimize the cost of outliers.

**Results**

The use of the adaptive loss proved to be efficient for several visual tasks, such as with variational auto-encoders for image generation.

Those are results for an unsupervised monocular depth estimation. Note that the loss being replaced is, in that case, the measure between two stereo images in a pair. 

![image](https://user-images.githubusercontent.com/8530703/81506966-41b3c080-92fa-11ea-8049-254d91c8b53f.png)

![image](https://user-images.githubusercontent.com/8530703/81506988-555f2700-92fa-11ea-9b94-5b8086a4b8e4.png)
